### DISCUSSION TITLE
Inconsistent Prop Behavior in Chakra UI Container Variants

### DISCUSSION THREAD

**Discussion by ijlind** [chakra-ui/chakra-ui#8925] [General]
### Description

There is an inconsistency in how the `Container` component handles various props (such as `p` & `padding`, `m` & `margin`, `px` &  `paddingX`, etc.) when using different sizes or variants in Chakra UI. The issue occurs when the base theme is using responsive values. It seems that the props passed to the component do not take precedence if a different shorthand name is used than in the base theme **unless** the new value is also in a responsive format.

### Link to Reproduction

https://codesandbox.io/p/sandbox/349kkm

### Steps to reproduce

1. Create a custom theme with sizes/variants for the Container component
2. Define padding properties in the size/variant as shown below
  ```theme.ts

import { extendTheme } from '@chakra-ui/react'

export const theme = extendTheme({
  components: {
    Container: {
      variants: {
        test: {
          p: { base: 0, md: 0 },
        },
      },
    },
  },
});
```
3. Create a simple React component to test the different padding props
```App.tsx

import { Button, Container, Text, VStack } from '@chakra-ui/react'

export const App = () => {
  return (
    <VStack gap={10}>
      <Container p={10} variant="test">THIS WORKS</Container>
      <Container padding={10} variant="test">THIS DOES NOT WORK</Container>
      <Container padding={{ base: 10, md: 10 }} variant="test">THIS WORKS</Container>
    </VStack>
  )
}
```
4. Observe the behavior of padding in the different containers.

### Chakra UI Version

2.10.1 (Latest as of the time of writing)

### Browser

Google Chrome 129

### Operating System

- [X] macOS
- [ ] Windows
- [ ] Linux

### Additional Information

The expected behavior is that the padding, margin, and other shorthand props should consistently be overridden regardless of whether short or long-form props are used, and should respect the custom variant of the container. However, the behavior is inconsistent across different props as indicated by the checkmarks (✅) and crosses (❌) in the reproduction code.

-------
Thank you for looking into this issue, and I appreciate your time and effort in maintaining this fantastic library!

**Comment by ijlind** [comment]
Actually did some further testing about this and found out that you don't even have to use sizes/variants. Just adding a responsive value to the `baseStyle` is enough: https://codesandbox.io/p/sandbox/7kq3y8

I noticed this first in our app due to how it worked on sizes and variants which led me to believe it was related to how those are handles specifically.

**Comment by segunadebayo** [comment]
This design flaw in v2 can't be changed at the moment due to its breaking potential. The logic should expand shorthand props before merging the props so the result is deterministic. Right now, it's not.

I recommend using consistent prop styles shorthand everywhere or longhand everywhere, to keep things predictable. Alternatively, you can use `!important` or `!` (for short)

We've improved this significantly in the coming v3 release.

**Comment by segunadebayo** [comment]
v3 merges shorthand and longhand correctly. This was a v2 issue.
